src/fetch/FetchExample.re

```diff
@@ -0,0 +1,87 @@
+type dog = string;
+
+type arrayDogs = array(dog);
+
+type state = { dogs: arrayDogs, isLoading: bool, error: bool };
+
+type action =
+  | DogsFetch
+  | DogsFetched(arrayDogs)
+  | DogsFailedToFetch;
+
+module Decode = {
+  let dogs = json : arrayDogs =>
+  Json.Decode.(
+    json |> field("message", array(string))
+         |> Array.map(dog => dog)
+  );
+};
+
+let component = ReasonReact.reducerComponent("FetchExample");
+
+let make = _children => {
+  ...component,
+  initialState: () => { dogs: [||], isLoading: true, error: false },
+  reducer: (action, state) => 
+    switch action {
+    | DogsFetch =>
+        ReasonReact.SideEffects(self => 
+          Js.Promise.(
+            Fetch.fetch("https://dog.ceo/api/breeds/list")
+            |> then_(Fetch.Response.json)
+            |> then_(json => 
+              json
+                |> Decode.dogs
+                |> (dogs => {
```

Comment:
You still don't need curly braces here. `|> dogs => self.send(DogsFetched(dogs))` is fine